framework/Models/EnsembleModel.py

```diff
@@ -604,11 +558,12 @@ def _externalRun(self,inRun, jobHandler):
     inputKwargs = inRun[2]
     identifier = inputKwargs.pop('prefix')
     tempOutputs = {}
-    tempTargetEvaluations = {}
+    inRunTargetEvaluations = {}
 
     for modelIn in self.orderList:
-      self.tempTargetEvaluations[modelIn].reset()
-      tempTargetEvaluations[modelIn] = copy.copy(self.tempTargetEvaluations[modelIn])
+      # reset the DataObject for the projection
+      self.localTargetEvaluations[modelIn].reset()
+      inRunTargetEvaluations[modelIn] = copy.copy(self.localTargetEvaluations[modelIn])
```

Comment:
Why do we need two local data objects, since we're calling "reset" on the instance copy?  Does "local" ever get used elsewhere?

Reply:
The reason to create a local copy in the run method is because the EnsembleModel run method can be run in multi-threading and the DataObjects are not thread safe (multiple threads can not modify the DO at the same otherwise a crash might occur)